executor/src/constant_evaluator/mod.rs

```diff
@@ -94,17 +94,16 @@ fn generate_values<T: FieldElement>(
             } else {
                 e
             };
+            let fun = evaluator::evaluate(e, &mut symbols.clone()).unwrap();
             (0..degree)
                 .into_par_iter()
                 .map(|i| {
-                    let mut symbols = symbols.clone();
-                    let fun = evaluator::evaluate(e, &mut symbols).unwrap();
                     evaluator::evaluate_function_call(
-                        fun,
+                        fun.clone(),
```

Comment:
This is an `Arc` so it should be relatively cheap.

Reply:
If we want to optimize this further, we could avoid thread synchronization by having one copy per thread, but I guess we should rather compile to rust instead...